pkg/helper/client_helper.go

```diff
@@ -29,7 +29,7 @@ func UnmanagedClient(cl client.Client) Client {
 // CreateOwned is an helper function that creates an object, sets owner reference and writes info & errors logs
 func (c *Client) CreateOwned(ctx context.Context, obj client.Object) error {
 	log := log.FromContext(ctx)
-	c.SetChanged(true)
+	c.SetChanged(false)
```

Comment:
hmm .. that sounds wrong? When an object is being created, it means some reconciliation / change is happening